ns for fx: add basic testing on cuda

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65593

Summary:

Adds test cases that the three Numeric Suite Core APIs work
when the models are on cuda.  In particular:
1. create models and move them to cuda
2. add loggers (if applicable)
3. run data through (if applicable)
4. extract results

It works without code changes because a `Logger` object is
created without any device specific objects (they only get
added if a data is passed through). It's good to have this tested.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_loggers_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_shadow_loggers_cuda
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31160897](https://our.internmc.facebook.com/intern/diff/D31160897)

vkuzo committed:
ns for fx: add basic testing on cuda

Summary:

Adds test cases that the three Numeric Suite Core APIs work
when the models are on cuda.  In particular:
1. create models and move them to cuda
2. add loggers (if applicable)
3. run data through (if applicable)
4. extract results

It works without code changes because a `Logger` object is
created without any device specific objects (they only get
added if a data is passed through). It's good to have this tested.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_loggers_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_shadow_loggers_cuda
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31160897).

jerryzh168 commented on test/quantization/fx/test_numeric_suite_fx.py:
```diff
@@ -1854,6 +1855,54 @@ def test_shadow_loggers_preserve_qat_numerics(self):
         ref_shadow = mc_shadows_mp(datum)
         self.assertEqual(ref_fp32, ref_shadow)
 
+    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
```
nit: do we need multigpu for this test?

jerryzh168 reviewed (approved):
lgtm

vkuzo committed:
Update on "ns for fx: add basic testing on cuda"


Summary:

Adds test cases that the three Numeric Suite Core APIs work
when the models are on cuda.  In particular:
1. create models and move them to cuda
2. add loggers (if applicable)
3. run data through (if applicable)
4. extract results

It works without code changes because a `Logger` object is
created without any device specific objects (they only get
added if a data is passed through). It's good to have this tested.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_loggers_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_shadow_loggers_cuda
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31160897](https://our.internmc.facebook.com/intern/diff/D31160897)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31160897).
